### ISSUE TITLE
[BUG] PostgreSQL backup fails on Ubuntu due to permission issue

### ISSUE THREAD

**Issue by przemyslavic** [epiphany-platform/epiphany#2877]
**Describe the bug**
Backup of PostgreSQL component fails on Ubuntu due to permission issue.
```
2022-01-14T17:39:58.7763918Z[38;21m17:39:58 INFO cli.engine.ansible.AnsibleCommand - TASK [backup : Ascertain primary node] *****************************************
2022-01-14T17:39:59.3594109Z[31;21m17:39:59 ERROR cli.engine.ansible.AnsibleCommand - fatal: [ci-rev13azubflannel-postgresql-vm-0]: FAILED! => {"msg": "Failed to set permissions on the temporary files Ansible needs to create when becoming an unprivileged user (rc: 1, err: chown: changing ownership of '/var/tmp/ansible-tmp-1642181998.856204-1842-45651123540402/': Operation not permitted\nchown: changing ownership of '/var/tmp/ansible-tmp-1642181998.856204-1842-45651123540402/AnsiballZ_postgresql_info.py': Operation not permitted\n}). For information on working around this, see https://docs.ansible.com/ansible/become.html#becoming-an-unprivileged-user"}
2022-01-14T17:39:59.4260791Z[31;21m17:39:59 ERROR cli.engine.ansible.AnsibleCommand - fatal: [ci-rev13azubflannel-postgresql-vm-1]: FAILED! => {"msg": "Failed to set permissions on the temporary files Ansible needs to create when becoming an unprivileged user (rc: 1, err: chown: changing ownership of '/var/tmp/ansible-tmp-1642181998.8913646-1843-256672895146475/': Operation not permitted\nchown: changing ownership of '/var/tmp/ansible-tmp-1642181998.8913646-1843-256672895146475/AnsiballZ_postgresql_info.py': Operation not permitted\n}). For information on working around this, see https://docs.ansible.com/ansible/become.html#becoming-an-unprivileged-user"}
```

NOTE: **Please also verify the recovery playbook**

It's been tested in #2602 and there were no changes in PostgreSQL backup since then so I suppose it's strictly related to upgrade of Ubuntu to 20.04.

**How to reproduce**
1. execute `epicli apply` to deploy a cluster with PostgreSQL component enabled (2vms)
2. execute `epicli backup -b /build/directory -f /backup/config.yml` 

**Expected behavior**
Backup of PostgreSQL component should be successful

**Config files**
Backup config
```yaml
---
kind: configuration/backup
title: Backup Config
provider: aws
name: default
specification:
  components:
    load_balancer:
      enabled: true
    logging:
      enabled: true
    monitoring:
      enabled: true
    postgresql:
      enabled: true
    rabbitmq:
      enabled: true
    kubernetes:
      enabled: true
```

**Environment**
- Cloud provider: [all]
- OS: [Ubuntu]

**epicli version**: v1.3, develop

**Additional context**

---

**DoD checklist**

* [ ] Changelog updated (if affected version was released)
* [ ] COMPONENTS.md updated / doesn't need to be updated
* [ ] Automated tests passed (QA pipelines)
  * [ ] apply
  * [ ] upgrade
* [ ] Case covered by automated test (if possible)
* [ ] Idempotency tested
* [ ] Documentation updated / doesn't need to be updated
* [ ] All conversations in PR resolved
* [ ] Backport tasks created / doesn't need to be backported

**Comment by to-bar** [comment]
Solution applied in #2881 and #2883 has been reverted since pipelining is incompatible with sudo's requiretty setting. This is because of [a quirk of the Python interpreter](https://github.com/ansible/ansible/blob/4b753084/lib/ansible/plugins/connection/ssh.py#L542), which enters interactive mode automatically when you pipe in data from a tty. 

From Ansible doc:

> Pipelining, if supported by the connection plugin, reduces the number of network operations required to execute a module on the remote server, by executing many Ansible modules without actual file transfer. This can result in a very significant performance improvement when enabled. However this conflicts with privilege escalation (become). For example, when using ‘sudo:’ operations you must first disable ‘requiretty’ in /etc/sudoers on all managed hosts, which is why it is disabled by default. This option is disabled if ANSIBLE_KEEP_REMOTE_FILES is enabled. This is a global option, each connection plugin can override either by having more specific options or not supporting pipelining at all.

To be on safe side we should check sudo's config in preflight checks or ensure `requiretty` is not enabled, like in this example: https://github.com/ansible/ansible/pull/13200#issuecomment-397248720

By default, `requiretty` is not enabled since RHEL 7.3: https://bugzilla.redhat.com/show_bug.cgi?id=1020147

**Comment by przemyslavic** [comment]
The changes have been reverted and it has been fixed in #2895.
